fdb-relational-core/src/main/java/com/apple/foundationdb/relational/recordlayer/query/LogicalOperator.java

```diff
@@ -397,50 +400,15 @@ public static LogicalOperator generateSimpleSelect(@Nonnull Expressions output,
         });
         final var expandedOutput = output.expanded();
         SelectExpression selectExpression;
-
-        if (canAvoidProjectingIndividualFields(output, logicalOperators)) {
```

Comment:
Can you give an example of when we can and when we cannot build with `*`

Reply:
Addressed this with 04bab1f where it now looks at the expected new type and compares it to the underlying value's type, and then omits the extra RCV if the types are the same. This change was necessary in order to have the `__ROW_VERSION` not exposed during star expansion, which is (and has always been) the desired semantics of `select *`.